sphinx/setup_command.py

```diff
@@ -138,20 +138,14 @@ def finalize_options(self):
             self.source_dir = self._guess_source_dir()
             self.announce('Using source directory %s' % self.source_dir)
         self.ensure_dirname('source_dir')
-        if self.source_dir is None:
-            self.source_dir = os.curdir
-        self.source_dir = abspath(self.source_dir)
-        if self.config_dir is None:
-            self.config_dir = self.source_dir
-        self.config_dir = abspath(self.config_dir)
 
         self.ensure_string_list('builder')
         if self.build_dir is None:
             build = self.get_finalized_command('build')
             self.build_dir = os.path.join(abspath(build.build_base), 'sphinx')  # type: ignore
             self.mkpath(self.build_dir)  # type: ignore
-        self.build_dir = abspath(self.build_dir)
         self.doctree_dir = os.path.join(self.build_dir, 'doctrees')
+        # TODO(stephenfin): Is this necessary?
```

Comment:
It is not needed. Builder class creates doctreedir if not exist.
https://github.com/sphinx-doc/sphinx/blob/a1f13e30000cdeceef7fc29b2bf3d0f45d5750ca/sphinx/builders/__init__.py#L85

Reply:
I'll fix this in the next version

Reply:
Done